model/task_json.go

```diff
@@ -36,6 +36,16 @@ type TagContainer struct {
 	Tag string `bson:"_id" json:"tag"`
 }
 
+type TagMeta struct {
+	Created  time.Time `bson:"created" json:"created"`
+	Revision string    `bson:"revision" json:"revision"`
+}
+
+type Tag struct {
+	Name string  `bson:"_id" json:"name"`
+	Meta TagMeta `bson:"obj" json:"obj"`
+}
```

Comment:
Can we prefix both of these types with "TaskJSON" and add the bsonutil.MustHaveTags for their fields?

Reply:
I already saw this pattern in this file. Since file name is `task_json.go` there is not reason to mention it twice, I think. However, the file belongs to package `model`. Can we move it to dedicated package?
I still think `TaskJSON.Tag` is better than `model.TaskJSONTag`. And it also looks weird when all variables are prefixed with `TaskJSON`. Just my opinion. 
The second question, what do you mean by `add the bsonutil.MustHaveTags`? Do you mean I should create a variable for each struct field? Also, why do we need these runtime assertations for tags? Doesn't such structs always have `bson` tag?

Reply:
Lots of small seperate packages is a weakness, and makes compilation more complicated than it needs to be. This is a weakness of the existing system. I think naming the types "TaskJSON" is a weakness (we can/should call them "TaskData" but that's awkward as well. Thankfully the type names are not super important, but leaving overly generic type names like "Tag" would be a problem. 

Please make this change. 

The bson.MustHaveTag function panics if a structure does not have a bson struct tag on start up, which is about the only mechanism we have to make sure we don't accidentally push junk data into the DB. Its awkward, and I'd like to have a different mechanism, but for now this is what we have.